Mathlib/Algebra/Homology/HomologySequence.lean

```diff
@@ -185,13 +185,13 @@ lemma opcycles_right_exact (S : ShortComplex (HomologicalComplex C c)) (hS : S.E
   have : Epi (ShortComplex.map S (eval C c i)).g := by dsimp; infer_instance
   have hj := (hS.map (HomologicalComplex.eval C c i)).gIsCokernel
   apply ShortComplex.exact_of_g_is_cokernel
-  refine' CokernelCofork.IsColimit.ofπ' _ _  (fun {A} k hk => by
+  refine CokernelCofork.IsColimit.ofπ' _ _  (fun {A} k hk => by
```

Comment:
I think it would be a good idea to replace the `refine`s with `exact`s where possible. (I can help with that if needed). Also, a `refine` within a `refine` seems like code smell.

Reply:
Replacing terminal `refine`s with `exact` turned out to result in slowing mathlib.  There is a PR somewhere with timings and all.

Nested `refine`s: I agree, but I think that it is a separate issue to replacing `refine'` with `refine`.

Reply:
I think both issues can be deal with in separate PRs.

Reply:
For completeness, #11890 is the PR that systematically replaced every terminal `refine` with `exact`.  You can see there that the modification made mathlib slower overall.  I then selected the files that were *improved* by the modification and PRed those alone.

So, unless there is the desire for a further "terminal `refine` audit", I would be skeptical of mass-replacing finishing `refine`s by `exact`.

Reply:
Interesting. I would not have expected `exact` to be slower than `refine`.

Reply:
I was also surprised by this, but luckily the PR was benchmarked before deciding to merge!